server/src/routes/asiv/public.ts

```diff
@@ -0,0 +1,46 @@
+import { Router, Request, Response } from "express";
+import axios from "axios";
+
+import { getTransactionInfo, processPayment } from "../../services/asiv";
+import { AsivTransactionInfo, FLORA_API_KEY } from "../../utils";
+
+const router: Router = Router();
+
+// GET request - get the payment page
+router.get("/payments/:tid", async (req: Request, res: Response) => {
+  const tid: string = req.params.tid;
+  const info: AsivTransactionInfo = getTransactionInfo(tid);
+
+  return res.render("index", {
+    layout: false,
+    paymentTo: "Flora",
+    userId: info.uid,
+    productId: info.pid,
+    productCost: info.pcost,
+  });
+});
+
+// POST request - make a payment
+router.post("/payments/:tid", async (req: Request, res: Response) => {
+  const tid: string = req.params.tid;
+
+  // try to process the payment
+  const info: AsivTransactionInfo = processPayment(tid);
+
```

Comment:
@Yonggiee halp add at line 29 the update of the user credits to reflect what he spent. The `info.pcost` is the amount that he is spending.

Reply:
To be tracked in #9.